Source/WebCore/page/scrolling/ScrollingCoordinatorTypes.h

```diff
@@ -116,23 +116,33 @@ enum class ViewportRectStability {
 
 enum class ScrollRequestType : uint8_t {
     PositionUpdate,
+    DeltaUpdate,
     CancelAnimatedScroll
 };
 
 struct RequestedScrollData {
     ScrollRequestType requestType { ScrollRequestType::PositionUpdate };
-    FloatPoint scrollPosition;
+    std::variant<FloatPoint, FloatSize> scrollPositionOrDelta;
     ScrollType scrollType { ScrollType::User };
     ScrollClamping clamping { ScrollClamping::Clamped };
     ScrollIsAnimated animated { ScrollIsAnimated::No };
-    std::optional<std::tuple<FloatPoint, ScrollType, ScrollClamping>> requestedDataBeforeAnimatedScroll { };
+    std::optional<std::tuple<ScrollRequestType, std::variant<FloatPoint, FloatSize>, ScrollType, ScrollClamping>> requestedDataBeforeAnimatedScroll { };
 
     void merge(RequestedScrollData&&);
+    WEBCORE_EXPORT FloatPoint destinationPosition(const FloatPoint&) const;
+    bool comparePositionOrDelta(const RequestedScrollData& other) const
```

Comment:
"compare" doesn't tell the reader if a true return value means "matched" or "mismatched", so it would be better to name it `positionOrDeltaEqual()`